core/templates/rid_owner.h

```diff
@@ -350,7 +350,7 @@ class RID_Alloc : public RID_AllocBase {
 				if (validator & 0x80000000) {
 					continue; //uninitialized
 				}
-				if (validator != 0xFFFFFFFF) {
+				if (chunks != nullptr && validator != 0xFFFFFFFF) {
```

Comment:
This should not be necessary, `chunks` is `null` only if `max_alloc == 0` (in this case, for loop will be skipped) or `memrealloc` failed (in this case, it will crash in `_allocate_rid()`).

Reply:
In this case, PVS-Studio probably want's us to check return value of `memrealloc` on the line `94`, and do something if it's `nullptr`. This is technically valid, but in practice if we failed to allocate 64 bytes it's not much we can do, except `CRASH_COND_MSG(!chunks, "Out of memory");`.

> On second thought maybe the error shows up because the actual null check below is unnecessary?

No, this check is valid, this code is reachable if no RIDs were allocated.

Reply:
Oh yes my bad, misread the indentation and thought the freeing was inside the check